src/main/config/config_eeprom.c

```diff
@@ -509,7 +509,7 @@ void writeConfigToEEPROM(void)
     bool success = false;
     // write it
     for (int attempt = 0; attempt < 3 && !success; attempt++) {
-        if (writeSettingsToEEPROM()) {
+        if (writeSettingsToEEPROM() && isEEPROMVersionValid() && isEEPROMStructureValid()) {
             success = true;
```

Comment:
```suggestion
        if (writeSettingsToEEPROM()) {
            success = isEEPROMVersionValid() && isEEPROMStructureValid();
```

Reply:
(@haslinghuis : your suggestion will call loadEEPROMFromExternalFlash when not necessary)